cli/src/pcluster/validators/networking_validators.py

```diff
@@ -57,6 +64,55 @@ def _validate(self, subnet_ids: List[str]):
             self._add_failure(str(e), FailureLevel.ERROR)
 
 
+class QueueSubnetsValidator(Validator):
+    """
+    Queue Subnets validator.
+
+    Check that there is no duplicate subnet id in the subnet_ids list.
+    Check that subnets in a queue belong to different AZs (EC2 Fleet requests do not support multiple subnets
+    in the same AZ).
+    """
+
+    @staticmethod
+    def _find_azs_with_multiple_subnets(az_subnet_ids_mapping):
+        return {az: subnet_ids for az, subnet_ids in az_subnet_ids_mapping.items() if len(subnet_ids) > 1}
+
+    def _validate(self, queue_name, subnet_ids: List[str], az_subnet_ids_mapping: dict):
+
+        # Test if there are duplicate IDs in subnet_ids
+        if len(set(subnet_ids)) < len(subnet_ids):
+            duplicate_ids = [subnet_id for subnet_id, count in Counter(subnet_ids).items() if count > 1]
+            self._add_failure(
+                "The following subnet ids are specified multiple times "
+                "in queue {0}: {1}.".format(
+                    queue_name,
+                    ", ".join(duplicate_ids),
+                ),
+                FailureLevel.ERROR,
+            )
+
+        # Test if the subnets are all in different AZs
```

Comment:
Are we sure these tests are mutually exclusive ?

It seems possible to have duplicated ID in the list of subnets and also have more than one subnet-id in the same AZ.

Reply:
No, they are not mutually exclusive. I just don't want the duplicate subnet IDs to trigger also the second part of the validator. Without this `if ... else` the duplicate subnet IDs would also be counted as multiple subnets in the same AZ, which to me doesn't make sense.

Reply:
Good catch @NSsirena. I changed implementation so that now it's possible to have both validation messages at the same time.